timeboost-sequencer/src/decrypt.rs

```diff
@@ -692,50 +689,42 @@ impl Worker {
             trace!(node = %self.label, from=%src, %res.committee_id, "not current committee");
             return Ok(());
         }
-
-        let WorkerState::DkgPending(ref mut subsets) = self.state else {
-            trace!(node = %self.label, from=%src, %res.committee_id, "not in a pending state");
-            return Ok(());
+        let subsets = match &mut self.state {
+            WorkerState::DkgPending(subsets) | WorkerState::HandoverPending(subsets) => subsets,
+            _ => {
+                trace!(node = %self.label, from=%src, %res.committee_id, "not in a pending state");
+                return Ok(());
+            }
         };
 
         let guard = self.key_stores.read();
-        let Some(key_store) = guard.get(res.committee_id) else {
-            return Err(DecrypterError::NoCommittee(res.committee_id));
-        };
-        key_store
+        let current = guard
+            .get(res.committee_id)
+            .ok_or(DecrypterError::NoCommittee(res.committee_id))?;
+
+        let prev = (guard.len() == 2).then(|| guard.last().clone());
```

Comment:
is this correct? node will only have 2 keystore when it's running `Running` state and `NextCommittee` event happens and a new keystore is registered and resharing starts, right? In this case, `prev` should be `guard.first()` instead?

also when a node just started (i.e. in `HandoverPending` state), it doesn't really have `prev`, correct? (since it only store the current keystore locally)

Reply:
> , last() should always return the oldest.

that's not true? [`last()`](https://docs.rs/arrayvec/latest/arrayvec/struct.ArrayVec.html#method.last) gave the most recently added. 

> In my view, a newly started node (that wasn’t part of the initial epoch) will be initialized with the previous committee’s description,

i see, this is an implicit decision that I don't know about. 
thanks for clarifying

Reply:
I guess you can merge for now and if this logic is wrong we should detect it during your next PR when doing the integration test.